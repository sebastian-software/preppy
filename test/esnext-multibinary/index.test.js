/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"

import pify from "pify"
import rimraf from "rimraf"
import execa from "execa"

import { preppy } from "../util"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)


test("Multi Binary from ESNext", async () => {
  await lazyDelete(resolve(__dirname, "./bin"))
  await preppy({
    cwd: __dirname
  })

  expect(await lazyRead(resolve(__dirname, "bin/first-cli.js"), "utf8")).toMatchSnapshot(
    "first"
  )
  expect(await lazyRead(resolve(__dirname, "bin/second-cli.js"), "utf8")).toMatchSnapshot(
    "second"
  )
})
