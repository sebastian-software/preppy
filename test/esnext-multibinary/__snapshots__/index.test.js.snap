// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multi Binary from ESNext: first 1`] = `
"#!/usr/bin/env node

/*! esnext-multibinary v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

var _snakeCase = require('lodash/snakeCase');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e[\\"default\\"] : e; }

var _snakeCase__default = /*#__PURE__*/_interopDefaultLegacy(_snakeCase);

const realWords = \\"Eine dicke Kuh lacht\\";

console.log(\\"Magic:\\", _snakeCase__default(realWords));
//# sourceMappingURL=first-cli.js.map
"
`;

exports[`Multi Binary from ESNext: second 1`] = `
"#!/usr/bin/env node

/*! esnext-multibinary v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

var _camelCase = require('lodash/camelCase');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e[\\"default\\"] : e; }

var _camelCase__default = /*#__PURE__*/_interopDefaultLegacy(_camelCase);

const realWords = \\"Eine dicke Kuh lacht\\";

console.log(\\"Magic:\\", _camelCase__default(realWords));
//# sourceMappingURL=second-cli.js.map
"
`;
